### PR TITLE
[python-api] - do not use core context menus for WindowXML containers

### PR DESCRIPTION
**Pull request by phil65**
@tamland  
This fixes the issues I mentioned in https://github.com/xbmc/xbmc/pull/9595 
Core context menu doesnt really work with listitems added to WindowXML instances, so let script authors deal with that via onAction().

### CODE REVIEW

**Comment by tamland** [comment]
im guessing this has something to do with the CGUIMediaWindow 'Interceptor'. So every time media window is changed, it breaks..

**Comment by phil65** [comment]
yup, correct, anything which updates the container of the mediawindow breaks it.
I guess to fix all issues, I also have to intercept MSG_GUI_UPDATE ? That one also breaks the current listing. (happens when addons get updated, or by manually triggering the builltin "Container.Refresh").
I know that this is all quite hacky, but I dont see any better way to get that stuff into a usable shape (I guess I am the only one who uses these parts at all :) )

**Comment by tamland** [comment]
Yep. I'll trust you with this. Not touching that class with a ten foot pole:)

**Comment by MartijnKaijser** [comment]
/me hands over an 11 foot pole ;)

**Comment by phil65** [comment]
I now completely intercept GUI_MSG_NOTIFY_ALL, which fixes the remaining issues I had.
@tamland , does that look fine enough from 10 foot away? :)

File: xbmc/interfaces/legacy/WindowXML.cpp [hunk 1]
```diff
@@ -319,6 +319,13 @@ namespace XBMCAddon
         }
         break;
 
+      case GUI_MSG_NOTIFY_ALL:
+        {
+          // GUI_MSG_NOTIFY_ALL breaks container content, so intercept it.
+          return true;
+        }
+        break;
```
Comment (tamland) [inline xbmc/interfaces/legacy/WindowXML.cpp]: no need for break and the brackets

File: xbmc/interfaces/legacy/WindowXML.cpp [hunk 2]
```diff
@@ -370,6 +377,9 @@ namespace XBMCAddon
                 PulseActionEvent();
                 return true;
               }
+              // the core context menu can lead to all sort of issues right now when used with WindowXMLs, so lets intercept the corresponding message
+              else if (controlClicked->IsContainer() && message.GetParam1() == ACTION_CONTEXT_MENU)
+                  return true;
```
Comment (tamland) [inline xbmc/interfaces/legacy/WindowXML.cpp]: double indent

**Comment by tamland** [comment]
Other than cosmetics, it looks ok ..from a distance

**Comment by phil65** [comment]
took care of cosmetics, thx for review.
jenkins build this please.

**Pull request merged**